internal/engine/extractors.go

```diff
@@ -16,3 +19,64 @@ func extractImageAndK8sTargets(specs []model.TargetSpec) (iTargets []model.Image
 	}
 	return iTargets, kTargets
 }
+
+// Check to see if we have k8s targets.
+func extractK8sTargets(specs []model.TargetSpec) []model.K8sTarget {
+	kTargets := make([]model.K8sTarget, 0)
+	for _, spec := range specs {
+		t, ok := spec.(model.K8sTarget)
+		if !ok {
+			continue
+		}
+		kTargets = append(kTargets, t)
+	}
+	return kTargets
+}
+
+// Extract image targets iff they can be updated in-place in a container.
+func extractImageTargetsForLiveUpdates(specs []model.TargetSpec, stateSet store.BuildStateSet) ([]model.ImageTarget, error) {
+	iTargets := make([]model.ImageTarget, 0)
+	for _, spec := range specs {
+		iTarget, ok := spec.(model.ImageTarget)
+		if !ok {
+			continue
+		}
+
+		state := stateSet[iTarget.ID()]
+		if state.IsEmpty() {
+			return nil, RedirectToNextBuilderf("In-place build does not support initial deploy")
+		}
+
+		// If this image doesn't need to be built at all, we can skip it.
+		if !state.NeedsImageBuild() {
+			continue
+		}
+
+		fbInfo := iTarget.MaybeFastBuildInfo()
+		if fbInfo == nil {
+			return nil, RedirectToNextBuilderf("In-place build only supports FastBuilds")
+		}
+
+		// Otherwise, manifest has already been deployed; try to update in the running container
```

Comment:
sigh, we have name collision with deployInfo = info needed to deploy vs. deployInfo = info about the existing deployment :/ comment suggestion below to clarify a little
```suggestion
		// Otherwise, manifest has already been deployed; IF we have info about the running container, update in place
```

(you might also clarify the err message at ln 63)

Reply:
ya, this should really be called ContainerInfo or something. updated the comments + error message